pkg/client/v1/client.go

```diff
@@ -0,0 +1,193 @@
+	"github.com/textileio/go-tableland/pkg/client"
+	"github.com/textileio/go-tableland/pkg/nonce/impl"
+	"github.com/textileio/go-tableland/pkg/parsing"
+	parserimpl "github.com/textileio/go-tableland/pkg/parsing/impl"
+	"github.com/textileio/go-tableland/pkg/tables/impl/ethereum"
+	"github.com/textileio/go-tableland/pkg/wallet"
+)
+
+var defaultChain = client.Chains.PolygonMumbai
+
+// Client is the Tableland client.
+type Client struct {
+	tblHTTP     *http.Client
+	tblContract *ethereum.Client
+	chain       client.Chain
+	wallet      *wallet.Wallet
+	parser      parsing.SQLValidator
+	baseURL     *url.URL
+}
+
+type config struct {
+	chain           *client.Chain
+	infuraAPIKey    string
+	alchemyAPIKey   string
+	local           bool
+	contractBackend bind.ContractBackend
+}
+
+// NewClientOption controls the behavior of NewClient.
+type NewClientOption func(*config)
+
+// NewClientChain specifies chaininfo.
+func NewClientChain(chain client.Chain) NewClientOption {
+	return func(ncc *config) {
+		ncc.chain = &chain
+	}
+}
+
+// NewClientInfuraAPIKey specifies an Infura API to use when creating an EVM backend.
+func NewClientInfuraAPIKey(key string) NewClientOption {
+	return func(c *config) {
+		c.infuraAPIKey = key
+	}
+}
+
+// NewClientAlchemyAPIKey specifies an Alchemy API to use when creating an EVM backend.
+func NewClientAlchemyAPIKey(key string) NewClientOption {
+	return func(c *config) {
+		c.alchemyAPIKey = key
+	}
+}
+
+// NewClientLocal specifies that a local EVM backend should be used.
+func NewClientLocal() NewClientOption {
+	return func(c *config) {
+		c.local = true
+	}
+}
+
+// NewClientContractBackend specifies a custom EVM backend to use.
+func NewClientContractBackend(backend bind.ContractBackend) NewClientOption {
+	return func(c *config) {
+		c.contractBackend = backend
+	}
+}
+
+// NewClient creates a new Client.
+func NewClient(ctx context.Context, wallet *wallet.Wallet, opts ...NewClientOption) (*Client, error) {
+	config := config{chain: &defaultChain}
+	for _, opt := range opts {
+		opt(&config)
+	}
+
+	contractBackend, err := getContractBackend(ctx, config)
+	if err != nil {
+		return nil, fmt.Errorf("getting contract backend: %v", err)
+	}
+
+	tblContract, err := ethereum.NewClient(
+		contractBackend,
+		tableland.ChainID(config.chain.ID),
+		config.chain.ContractAddr,
+		wallet,
+		impl.NewSimpleTracker(wallet, contractBackend),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("creating contract client: %v", err)
+	}
+
+	parserOpts := []parsing.Option{
+		parsing.WithMaxReadQuerySize(35000),
+		parsing.WithMaxWriteQuerySize(35000),
+	}
+
+	parser, err := parserimpl.New([]string{
+		"sqlite_",
+		systemimpl.SystemTablesPrefix,
+		systemimpl.RegistryTableName,
+	}, parserOpts...)
+	if err != nil {
+		return nil, fmt.Errorf("new parser: %s", err)
+	}
+
+	parser, err = parserimpl.NewInstrumentedSQLValidator(parser)
+	if err != nil {
+		return nil, fmt.Errorf("instrumenting parser: %s", err)
+	}
```

Comment:
Does it make sense to use a instrumented parser here? What does instrumentation do if the client is being used in a client side program (not within the validator)?

Reply:
Yeah, might be too much to think about collecting metrics for a client.
I'll remove the instrumentation :+1: